pkg/sdn/plugin/registry.go

```diff
@@ -263,6 +263,34 @@ func (registry *Registry) GetEgressNetworkPolicies() ([]osapi.EgressNetworkPolic
 	return policyList.Items, nil
 }
 
+// Returns true to indicate status update is needed
+type UpdateNodeFunc func(node *kapi.Node) bool
+
+func (registry *Registry) UpdateNodeStatus(node *kapi.Node, updateFunc UpdateNodeFunc) (*kapi.Node, error) {
```

Comment:
> This is an odd helper/abstraction… I'd just do it directly

@liggitt same here; the indirectness is a result of the encapsulation of the osClient/kClient being in the Registry object.  Need input from @danwinship @pravisankar @rajatchopra to see whether we can/should start deconstructing the registry.